pkg/util/utils_test.go

```diff
@@ -108,3 +111,40 @@ Error 2`
 		t.Fatalf("Unexpected error: %s", err)
 	}
 }
+
+type URLHandler200 struct{}
+
+func (h *URLHandler200) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "Hello World")
+}
+
+type URLHandler400 struct{}
+
+func (h *URLHandler400) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	http.Error(w, "failed to write report", 400)
+}
+
+func TestReportError(t *testing.T) {
+	handler200 := &URLHandler200{}
+	server := httptest.NewServer(handler200)
```

Comment:
You could move this inline to save on having to define the function and struct above. Something like:

```golang
server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
  fmt.Fprintf(w, "Hello, world!")
})